iree/tools/CMakeLists.txt

```diff
@@ -478,6 +478,21 @@ if(${IREE_BUILD_COMPILER})
     RUNTIME DESTINATION bin
   )
 
+  if(NOT LLD_SYMLINKS_TO_CREATE)
+  set(LLD_SYMLINKS_TO_CREATE
+    lld-link ld.lld ld64.lld wasm-ld)
+  endif()
+
+  foreach(link ${LLD_SYMLINKS_TO_CREATE})
+    install(
+      CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink lld ${link} \
```

Comment:
What are the intended outputs and uses of this?

On Windows I get these:

```
[build] -- Up-to-date: D:/dev/projects/iree-build/install/bin/lld.exe
[build] -- Create symlink: bin/lld-link -> bin/lld
[build] -- Create symlink: bin/ld.lld -> bin/lld
[build] -- Create symlink: bin/ld64.lld -> bin/lld
[build] -- Create symlink: bin/wasm-ld -> bin/lld
```

I can't run the `lld.exe` program directly:
```
λ D:\dev\projects\iree-build\install\bin\lld.exe
lld is a generic driver.
Invoke ld.lld (Unix), ld64.lld (macOS), lld-link (Windows), wasm-ld (WebAssembly) instead
```

(the `iree-build\third_party\llvm-project\llvm\bin` directory _does_ have a `lld-link.exe` that works, but that is not installed into `iree-build/install/bin` right now)

and I might be missing something, but I can't run the symlinks either:
```
λ D:\dev\projects\iree-build\install\bin\lld-link
'D:\dev\projects\iree-build\install\bin\lld-link' is not recognized as an internal or external command,
operable program or batch file.
```

Reply:
Can't we just use what LLVM already provides for this?

https://github.com/llvm/llvm-project/blob/main/lld/tools/lld/CMakeLists.txt
https://github.com/llvm/llvm-project/blob/main/lld/cmake/modules/AddLLD.cmake
https://github.com/llvm/llvm-project/blob/main/llvm/cmake/modules/AddLLVM.cmake

They already have the platform logic implemented, and anything custom we do here will be brittle and difficult to maintain.

Reply:
Those macros requires `lld` to be a target built under the same directory. If I use those macros, I will get
```
TARGET 'lld' was not created in this directory.
```